internal/grpc/services/appprovider/appprovider.go

```diff
@@ -93,18 +108,80 @@ func getProvider(c *config) (app.Provider, error) {
 	}
 }
 
-func (s *service) Open(ctx context.Context, req *providerpb.OpenRequest) (*providerpb.OpenResponse, error) {
-	iframeLocation, err := s.provider.GetIFrame(ctx, req.ResourceInfo.Id, req.AccessToken)
+func (s *service) OpenFileInAppProvider(ctx context.Context, req *providerpb.OpenFileInAppProviderRequest) (*providerpb.OpenFileInAppProviderResponse, error) {
+
+	log := appctx.GetLogger(ctx)
+
+	wopiurl := s.conf.WopiURL
+	iopsecret := s.conf.IopSecret
+	storageID := s.conf.StorageID
+	folderURL := s.conf.UIURL + filepath.Dir(req.Ref.GetPath())
+
+	httpClient := rhttp.GetHTTPClient(
+		rhttp.Context(ctx),
+		// TODO make insecure configurable
+		rhttp.Insecure(true),
+		// TODO make timeout configurable
+		rhttp.Timeout(time.Duration(24*int64(time.Hour))),
+	)
+	httpReq, err := rhttp.NewRequest(ctx, "GET", wopiurl+"wopi/iop/open", nil)
```

Comment:
Shouldn't this be carried out at the driver level, where this is a driver for wopi? Other apps would not have the exact same endpoint

Reply:
After the discussion with @glpatcern I think the driver is for a later stage.